io-engine/src/core/block_device.rs

```diff
@@ -82,6 +85,14 @@ pub trait BlockDevice {
         &self,
         listener: DeviceEventSink,
     ) -> Result<(), CoreError>;
+
+    fn is_zoned(&self) -> bool;
```

Comment:
This should probably be part of a separate trait which can be made as dependency in the BlockDevice trait?
trait BlockDevice: ZonedBlockDevice

Reply:
I've the same suggestion.